### PR TITLE
Feature/rails 5.2

### PR DESCRIPTION
**Pull request by bricesanchez**
We support Rails 5.2, we need to have a green build to merge this PR.

We will have to wait for rails 5.2.0.rc2, see: https://github.com/refinery/refinerycms/pull/3352#discussion_r173149307

[UPDATE 2018/06/02]:
* [x] Wait for this trivial merge: https://github.com/parndt/decorators/pull/17

### CODE REVIEW

**Comment by bricesanchez** [comment]
Hi @parndt or @simi !

Could you help me to find why we have this error?
```
Failures:

  1) Refinery::Page custom slugs given a page with a custom_slug exists fails validation when a new record uses that custom_slug
     Failure/Error: new_page.valid?

     NoMethodError:
       undefined method `except!' for nil:NilClass
     # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/globalize-5.1.0/lib/globalize/active_record/adapter_dirty.rb:45:in `_reset_attribute'
     # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/globalize-5.1.0/lib/globalize/active_record/adapter_dirty.rb:16:in `write'
     # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/globalize-5.1.0/lib/globalize/active_record/instance_methods.rb:31:in `write_attribute'
     # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/globalize-5.1.0/lib/globalize/active_record/class_methods.rb:96:in `block in define_translated_attr_writer'
     # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/bundler/gems/friendly_id-80bc0022910e/lib/friendly_id/slugged.rb:380:in `unset_slug_if_invalid'
     # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activemodel-5.2.0.rc1/lib/active_model/validations/callbacks.rb:118:in `run_validations!'
     # /Users/bricesanchez/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activemodel-5.2.0.rc1/lib/active_model/validations.rb:339:in `valid?'
     # ./pages/spec/models/refinery/page_url_spec.rb:278:in `block (4 levels) in <module:Refinery>'
```

Also, should we support Rails 5.1 and 5.2 on Refinery 4.1? Drop support of Rails 5.1 for Refinery 4.1? Start using Appraisal gem or wait for a request from the community?

File: core/refinerycms-core.gemspec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 require File.expand_path('../../core/lib/refinery/version', __FILE__)
 
 version = Refinery::Version.to_s
-rails_version = ['>= 5.1.0', '< 5.2']
+rails_version = ['>= 5.2.0.rc1', '< 6']
```
Comment (parndt) [inline core/refinerycms-core.gemspec]: seems like this could stay at `>= 5.1.0` and you could specify the rc1 in the Gemfile

Comment (bricesanchez) [reply core/refinerycms-core.gemspec]: At this time the only breaking change between Rails 5.1 and 5.2 is this : https://github.com/refinery/refinerycms/pull/3352/commits/70cd41cb09afec28ff65fc4b02dd636a9a201591 

So i'm ok to do what to say.

Any thoughts for this last remaining error `undefined method `except!' for nil:NilClass` ?

Comment (bricesanchez) [reply core/refinerycms-core.gemspec]: Oh it looks like it's a rc1 bug, `mobility` gem has also this kind of problem: 
https://github.com/shioyama/mobility/issues/164
https://github.com/rails/rails/commit/5fcbdcfb574c731841be12764c50d9587b58345f

Comment (parndt) [reply core/refinerycms-core.gemspec]: seems like it might be addressed in the next rc

**Comment by shioyama** [comment]
FWIW, released Mobility 0.5.1 with a fix for 5.2.0.rc2. The problematic change in rc1 was [reverted](https://github.com/rails/rails/commit/5fcbdcfb574c731841be12764c50d9587b58345f).

**Comment by parndt** [comment]
Thanks @shioyama !!

**Comment by bricesanchez** [comment]
@parndt  ready to merge?

**Comment by bricesanchez** [comment]
🆙 @parndt 🙃

**Comment by ttmthanh** [comment]
@bricesanchez But... What about 'refinerycms-authentication-devise' and 'refinerycms-inquiries' ? Because when I update rails and refinerycms, it has a bit of a conflict.

**Comment by bricesanchez** [comment]
@ttmthanh You could use the master branch for `refinerycms-inquiries`:
https://github.com/refinery/refinerycms-inquiries

For `refinerycms-authentication-devise`, i've forget to ask Philip for a review of this PR:
https://github.com/refinery/refinerycms-authentication-devise/pull/40

**Comment by ttmthanh** [comment]
@bricesanchez Please help me ask him :D Thank you very much :+1: :disappointed:

**Comment by ttmthanh** [comment]
`gem 'refinerycms-authentication-devise', git: 'https://github.com/refinery/refinerycms-authentication-devise', branch: 'master'
`
So I could use like this, right? :smiley:

**Comment by ttmthanh** [comment]
@bricesanchez Sorry sir... Please help me if you are free..

**Comment by parndt** [comment]
@ttmthanh I merged it..

**Comment by ttmthanh** [comment]
@parndt Thanks so much... Because I am trying to use the Active storage service (Rails 5.2) combined with refinerycms..

**Comment by bricesanchez** [comment]
> @parndt Thanks so much... Because I am trying to use the Active storage service (Rails 5.2) combined with refinerycms..

Nice! We just merged a bugfix with ActiveStorage and Refinery #3385 

> @parndt refinery/refinerycms-page-images#151 Sorry sir...

You will have to use the `feature/mobility` branch of `refinerycms-page-images` until we merge this PR

**Pull request merged**